### PR TITLE
Allow migration to be run within the UI

### PR DESCRIPTION
**Pull request by trackleft**
## Description
See migrations in UI here: https://bed5f77e-6261-4d6f-bdc9-ce08d378ddf0--site-migration.probo.build/admin/structure/migrate/manage/az_migration/migrations

Also refactored the dependency tree
Also added schema for az_migration module
Also removed dependency on az_carousel
Also enable az_carousel in probo migration config

The main problem with this approach comes at the contributor's expense.
Now, any time you update a migration yml file you must import config via config_distro

## Types of changes

**Arizona Quickstart** (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

**Drupal core**
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

**Drupal contrib projects**
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      type: string
+      label: 'HTTP Protocol to use.'
+    migrate_d7_private_path:
+      label: 'The private file directory path, if any.'
+      type: string
+      nullable: true
+    migrate_d7_temporary_path:
+      label: 'The temporary file directory path, if any.'
+      type: string
+      nullable: true
```
Comment (trackleft) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:22]:
```suggestion
      nullable: true

```

File: modules/custom/az_migration/config/optional/migrate_plus.migration.az_paragraph_content_marquee.yml [hunk 2]
```diff
@@ -117,3 +117,8 @@ dependencies:
   enforced:
     module:
       - az_migration
+  module:
+    - az_paragraphs_text_media
+migration_dependencies:
+  required:
+    - az_media
```
Comment (trackleft) [inline modules/custom/az_migration/config/optional/migrate_plus.migration.az_paragraph_content_marquee.yml:124]:
```suggestion
    - az_media

```

**Comment by camikazegreen** [comment]
This is pretty cool! 
I'm assuming the source site connection still needs to be done manually through settings.php? I didn't see a spot in the UI to add db credentials, etc.

**Comment by joeparsons** [comment]
Does this address #1389 or #1394 at all?

**Pull request closed without merging**